Mathlib/Topology/Algebra/Category/ProfiniteGrp.lean

```diff
@@ -161,4 +164,9 @@ instance : HasForget₂ ProfiniteGrp Grp where
     map := fun f => f.toMonoidHom
   }
 
+/-- A closed subgroup of a profinite group is profinite. -/
+def ofClosedSubgroup {G : ProfiniteGrp} (H : ClosedSubgroup G)  : ProfiniteGrp :=
+  letI : CompactSpace H := isCompact_iff_compactSpace.mp (IsClosed.isCompact H.isClosed')
+  of H.1
```

Comment:
I think it would be useful if this was a standalone instance in the `ClosedSubgroup` file for any `G` with `[CompactSpace G]`.

Reply:
I draged the instance out side, but that `letI` then `inferInstance` still cannot be deleted, I think that doesn't matters much, but can it be designed better to do so?